integration-tests/smoke/keeper_test.go

```diff
@@ -171,41 +165,48 @@ func getKeeperSuite(
-					// Let upkeep be performed atleast once
+				It("watches all the registered upkeeps perform and then cancels them from the registry", func() {
 					Eventually(func(g Gomega) {
-						cnt, err := consumer.Counter(context.Background())
-						g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's Counter shouldn't fail")
-						g.Expect(cnt.Int64()).Should(BeNumerically(">", int64(0)), "Expected consumer counter to be greater than 0, but got %d", cnt.Int64())
-						log.Info().Int64("Upkeep counter", cnt.Int64()).Msg("Upkeeps performed")
+						// Check if the upkeeps are performing by analysing their counters and checking they are greater than 0
+						for i := 0; i < len(upkeepIDs); i++ {
+							counter, err := consumers[i].Counter(context.Background())
+							g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's counter shouldn't fail")
+							g.Expect(counter.Int64()).Should(BeNumerically(">", int64(0)),
+								"Expected consumer counter to be greater than 0, but got %d", counter.Int64())
+							log.Info().Int64("Upkeep counter", counter.Int64()).Msg("Upkeeps performed")
+						}
 					}, "1m", "1s").Should(Succeed())
 
-					// Now cancel the upkeep as registry owner, it should get immediately cancelled
-					err := registry.CancelUpkeep(upkeepID)
-					Expect(err).ShouldNot(HaveOccurred(), "Upkeep should get cancelled successfully")
-					err = networks.Default.WaitForEvents()
-					Expect(err).ShouldNot(HaveOccurred(), "Error waiting for cancel upkeep tx")
+					// Cancel all the registered upkeeps via the registry
+					for i := 0; i < len(upkeepIDs); i++ {
+						err := registry.CancelUpkeep(upkeepIDs[i])
+						Expect(err).ShouldNot(HaveOccurred(), "Upkeep should get cancelled successfully")
+						err = networks.Default.WaitForEvents()
+						Expect(err).ShouldNot(HaveOccurred(), "Error waiting for cancel upkeep tx")
+					}
 
-					// Get existing performed count
-					existingCnt, err := consumer.Counter(context.Background())
-					Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's Counter shouldn't fail")
-					log.Info().Int64("Upkeep counter", existingCnt.Int64()).Msg("Upkeep cancelled")
-
-					// Expect count to be remain consistent
 					Consistently(func(g Gomega) {
-						cnt, err := consumer.Counter(context.Background())
-						g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's Counter shouldn't fail")
-						g.Expect(cnt.Int64()).Should(
-							Equal(existingCnt.Int64()),
-							"Expected consumer counter to to remain constant at %d, but got %d", existingCnt.Int64(), cnt.Int64(),
-						)
+						for i := 0; i < len(upkeepIDs); i++ {
+							// Obtain the amount of times the upkeep has been executed so far
+							counterAfterCancellation, err := consumers[i].Counter(context.Background())
+							Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's counter shouldn't fail")
+							log.Info().Int64("Upkeep counter", counterAfterCancellation.Int64()).Msg("Upkeep cancelled")
```

Comment:
You'll need this outside consistently, o/w counterAfterCancellation can change on each check